src/openms/source/CHEMISTRY/EnzymaticDigestion.cpp

```diff
@@ -171,20 +160,56 @@ namespace OpenMS
     log_model_threshold_ = threshold;
   }
 
+  std::vector<Size> EnzymaticDigestion::tokenise(const AASequence& protein) const
+  {
+    String s = protein.toUnmodifiedString();
+    boost::regex re(enzyme_.getRegEx());
+    boost::sregex_token_iterator i(s.begin(), s.end(), re, -1);
+    boost::sregex_token_iterator j;
+    std::vector<Size> positions;
+    Size pos = -1;
+    while (i != j)
+    {
+      AASequence pep = AASequence::fromString((String)(*i++));
```

Comment:
You are right. it was the code that I thought there might be modifications contained in the sequence. If AASequence::toUnmodifiedString is used, there should be no problem then.